<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>974</width>
    <height>1047</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>401</width>
      <height>221</height>
     </rect>
    </property>
    <property name="title">
     <string>4.1.1 字符串与数值之间的转换</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>30</y>
       <width>336</width>
       <height>176</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="v">
      <item>
       <layout class="QGridLayout" name="g">
        <item row="0" column="0">
         <widget class="QLabel" name="labelC">
          <property name="text">
           <string>数量</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="editNum">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLabel" name="labelA">
          <property name="text">
           <string>单价</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QLineEdit" name="editPrice">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="labelB">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="BtnCal">
          <property name="text">
           <string>计算</string>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QLineEdit" name="editTotal">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QLabel" name="labelD">
          <property name="text">
           <string>总价</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="v1"/>
      </item>
      <item>
       <layout class="QGridLayout" name="g1">
        <item row="0" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>十进制</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QPushButton" name="btnBin">
          <property name="text">
           <string>转换成其他进制</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>十六进制</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="DecimalEdit">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLineEdit" name="HexadecimalEdit">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QPushButton" name="Btn_16">
          <property name="text">
           <string>转换成其他进制</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QPushButton" name="btnDec">
          <property name="text">
           <string>转换成其他进制</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>二进制</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="BinaryEdit">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>10</y>
      <width>521</width>
      <height>351</height>
     </rect>
    </property>
    <property name="title">
     <string>4.1.2 QString的常用功能</string>
    </property>
    <widget class="QGroupBox" name="groupBox_4">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>80</y>
       <width>101</width>
       <height>181</height>
      </rect>
     </property>
     <property name="title">
      <string>数字</string>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>83</width>
        <height>141</height>
       </rect>
      </property>
      <layout class="QFormLayout" name="formLayout_3">
       <item row="0" column="0">
        <widget class="QPushButton" name="btnCount">
         <property name="text">
          <string>count</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="btnSize">
         <property name="text">
          <string>size</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QPushButton" name="btnIndex">
         <property name="text">
          <string>indexOf</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QPushButton" name="btnLastIndexOf">
         <property name="text">
          <string>lastIndexOf</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QCheckBox" name="checkBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>330</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>contains</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox">
     <property name="geometry">
      <rect>
       <x>430</x>
       <y>320</y>
       <width>42</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>380</x>
       <y>320</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>count()</string>
     </property>
    </widget>
    <widget class="QGroupBox" name="groupBox_5">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>80</y>
       <width>201</width>
       <height>181</height>
      </rect>
     </property>
     <property name="title">
      <string>逻辑判断</string>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>11</x>
        <y>21</y>
        <width>181</width>
        <height>141</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QPushButton" name="btnEndsWith">
         <property name="text">
          <string>endsWith</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="btnStartsWith">
         <property name="text">
          <string>startsWith</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="btnIsNull">
         <property name="text">
          <string>isNull</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="btnIsEmpty">
         <property name="text">
          <string>isEmpty</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QPushButton" name="btnContains">
         <property name="text">
          <string>contains</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QRadioButton" name="radioButton">
         <property name="text">
          <string>不区分大小写</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>80</y>
       <width>191</width>
       <height>181</height>
      </rect>
     </property>
     <property name="title">
      <string>字符串</string>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>158</width>
        <height>141</height>
       </rect>
      </property>
      <layout class="QFormLayout" name="f">
       <item row="0" column="0">
        <widget class="QPushButton" name="btnAppend">
         <property name="text">
          <string>Append</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="btnPrepend">
         <property name="text">
          <string>Prepend</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="btnToUpper">
         <property name="text">
          <string>toUpper</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="btnToLower">
         <property name="text">
          <string>toLower</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QPushButton" name="btnLeft">
         <property name="text">
          <string>left</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QPushButton" name="btnRight">
         <property name="text">
          <string>right</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QPushButton" name="btnSection">
         <property name="text">
          <string>section</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QPushButton" name="btnSimpli">
         <property name="text">
          <string>simplified</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QPushButton" name="btnTrimmed">
         <property name="text">
          <string>trimmed</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>501</width>
       <height>48</height>
      </rect>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>str1</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="str1_text">
        <item>
         <property name="text">
          <string>qwe</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>asd</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>zxc</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>123</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>宋炜</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>123.png</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>str2</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="str_text">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>11</x>
       <y>271</y>
       <width>501</width>
       <height>41</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>结果</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="resultText">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_6">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>230</y>
      <width>401</width>
      <height>221</height>
     </rect>
    </property>
    <property name="title">
     <string>4.2 SpinBox 的使用</string>
    </property>
    <widget class="QWidget" name="layoutWidget_5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>336</width>
       <height>176</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="v_7">
      <item>
       <layout class="QGridLayout" name="g_7">
        <item row="0" column="2">
         <widget class="QLabel" name="labelA_7">
          <property name="text">
           <string>单价</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="labelC_7">
          <property name="text">
           <string>数量</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="BtnCal_7">
          <property name="text">
           <string>计算</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="labelB_7">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QLabel" name="labelD_7">
          <property name="text">
           <string>总价</string>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QSpinBox" name="spbTotal"/>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="spbNum"/>
        </item>
        <item row="0" column="3">
         <widget class="QSpinBox" name="spbPrice"/>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="v1_7"/>
      </item>
      <item>
       <layout class="QGridLayout" name="g1_7">
        <item row="1" column="0">
         <widget class="QLabel" name="label_29">
          <property name="text">
           <string>二进制</string>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QPushButton" name="Btn_22">
          <property name="text">
           <string>转换成其他进制</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QPushButton" name="btnBin_7">
          <property name="text">
           <string>转换成其他进制</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_28">
          <property name="text">
           <string>十六进制</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QPushButton" name="btnDec_7">
          <property name="text">
           <string>转换成其他进制</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_27">
          <property name="text">
           <string>十进制</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="spbBin"/>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="spbDec"/>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="spbHex"/>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_7">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>370</y>
      <width>521</width>
      <height>541</height>
     </rect>
    </property>
    <property name="title">
     <string>4.3.1 实例功能</string>
    </property>
    <widget class="QGroupBox" name="groupBox_9">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>400</y>
       <width>331</width>
       <height>131</height>
      </rect>
     </property>
     <property name="title">
      <string>GroupBox</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QDial" name="dial"/>
      </item>
      <item>
       <widget class="QLCDNumber" name="lcdNumber"/>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_8">
        <property name="title">
         <string>LCD显示进制</string>
        </property>
        <widget class="QRadioButton" name="radioButton_2">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>20</y>
           <width>89</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>十进制</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="radioButton_3">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>40</y>
           <width>89</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>二进制</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="radioButton_4">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>80</y>
           <width>89</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>十六进制</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="radioButton_5">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>60</y>
           <width>89</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>八进制</string>
         </property>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QGroupBox" name="groupBox_10">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>411</width>
       <height>211</height>
      </rect>
     </property>
     <property name="title">
      <string>GroupBox</string>
     </property>
     <widget class="QTextEdit" name="textColorx">
      <property name="geometry">
       <rect>
        <x>300</x>
        <y>30</y>
        <width>101</width>
        <height>161</height>
       </rect>
      </property>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>40</y>
        <width>241</width>
        <height>108</height>
       </rect>
      </property>
      <layout class="QFormLayout" name="formLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Red</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSlider" name="redSd">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Green</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSlider" name="GreenSd">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Blue</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QSlider" name="BlueSd">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Alpha</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QSlider" name="AlpheSd">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_11">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>230</y>
       <width>401</width>
       <height>151</height>
      </rect>
     </property>
     <property name="title">
      <string>水平</string>
     </property>
     <widget class="QSlider" name="HorSd">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>22</y>
        <width>361</width>
        <height>22</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QProgressBar" name="progressBar">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>381</width>
        <height>20</height>
       </rect>
      </property>
      <property name="value">
       <number>24</number>
      </property>
     </widget>
     <widget class="QScrollBar" name="horizontalScrollBar">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>411</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_12">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>20</y>
       <width>71</width>
       <height>361</height>
      </rect>
     </property>
     <property name="title">
      <string>GroupBox</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QSlider" name="VerSd">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QProgressBar" name="progressBar_2">
        <property name="value">
         <number>24</number>
        </property>
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QScrollBar" name="verticalScrollBar">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>974</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
